site/e2e/tests/workspaces/autoCreateWorkspace.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { test, expect } from "@playwright/test";
+import { username } from "../../constants";
+import {
+  createTemplate,
+  createWorkspace,
+  echoResponsesWithParameters,
+} from "../../helpers";
+import { emptyParameter } from "../../parameters";
+import type { RichParameter } from "../../provisionerGenerated";
+
+test("create workspace in auto mode", async ({ page }) => {
+  const richParameters: RichParameter[] = [
+    { ...emptyParameter, name: "repo", type: "string" },
+  ];
+  const template = await createTemplate(
+    page,
+    echoResponsesWithParameters(richParameters),
+  );
+  const name = "test-workspace";
+  await page.goto(
+    `/templates/${template}/workspace?mode=auto&param.repo=example&name=${name}`,
+    {
+      waitUntil: "domcontentloaded",
+    },
+  );
+  await expect(page).toHaveTitle(`${username}/${name} - Coder`);
+});
+
+test("use an existing workspace that matches the `match` parameter instead of creating a new one", async ({
+  page,
+}) => {
+  const richParameters: RichParameter[] = [
+    { ...emptyParameter, name: "repo", type: "string" },
+  ];
+  const template = await createTemplate(
```

Comment:
I haven't worked with the E2E stuff much yet, so just to make sure I'm understanding: these tests _aren't_ isolated, right? That's how the second test is able to have a workspace already exist, even though it didn't explicitly make two of them?

Reply:
They are not isolated, but I ensure that a previous workspace with the right match exists in line 39.